frontend/public/components/utils/list-dropdown.jsx

```diff
@@ -78,13 +79,21 @@ class ListDropdown_ extends React.Component {
           unsortedList,
         );
       });
-
       const sortedList = {};
       _.keys(unsortedList)
         .sort()
         .forEach((key) => {
           sortedList[key] = unsortedList[key];
         });
+
+      if ((selectedKey === undefined || !_.isEmpty(selectedKey)) && selectFirst) {
```

Comment:
I think this condition should be enough.
```suggestion
      if (selectFirst) {
```

Reply:
Only selectFirst condition does not work as expected. We also have to check if the selectedKey is undefined or empty as the component re-renders multiple times and causes issues like blank dropdowns and inconsistent working